pavelib/edraak.py

```diff
@@ -6,111 +6,219 @@
 import polib
 from git import Repo
 import os
+import contextlib
+from shutil import copyfile
+
 from .utils.cmd import django_cmd
 
-PLATFORM_ROOT = path('.')
-git_repo = Repo(PLATFORM_ROOT)
+EDRAAK_RESOURCES = (
+    'edraak-platform',
+    'edraak-platform-2015-theme',
+)
+
+ARABIC_LOCALE_DIR = 'conf/locale/ar/LC_MESSAGES/'
+
+OTHER_RTL_LOCALES = ('eo', 'rtl', 'he', 'fa',)
+
+git_repo = Repo('.')
+
+
+@contextlib.contextmanager
+def working_directory(path):
+    """
+    A context manager which changes the working directory to the given
+    path, and then changes it back to its previous value on exit.
+    """
+    prev_cwd = os.getcwd()
+    os.chdir(path)
+    yield
+    os.chdir(prev_cwd)
+
+
+def js_nonjs(func):
+    """
+    Make a task run for both `django.po` and `djangojs.po` in a DRY way.
+    """
+    def wrapper():
+        """
+        Run it twice.
+        """
+        func(is_js=False, suffix='')
+        func(is_js=True, suffix='js')
+
+    # Keep the original name for Paver
+    wrapper.__name__ = func.__name__
 
-try:
-    from pygments.console import colorize
-except ImportError:
-    colorize = lambda color, text: text  # pylint: disable-msg=invalid-name
+    return wrapper
+
+
+def run_for_js_nonjs(func):
+    """
+    Actually run the function twice, rather than decorating the function.
+    """
+
+    func2 = js_nonjs(func)
+    func2()
 
 
 @task
-def edraak_transifex_pull():
+def clean_repo_check():
     # Start with clean translation state
-    clean_git_repo_msg = "The repo has local modifications. Please stash or commit your changes."
+    clean_git_repo_msg = (
+        'The repo has local modifications. '
+        'Please stash or commit your changes.'
+    )
+
     assert not git_repo.is_dirty(untracked_files=True), clean_git_repo_msg
 
+
+@task
+def i18n_transifex_pull_edraak():
```

Comment:
Why not naming it: `i18n_edraak_pull` instead? I think it would be more edxonic and straight forward. All of edX translations are managed by Transifex, so there's no need to specify it in translations commands to avoid understanding it as there's another organization dealing with translations together with Transifex.